### PR TITLE
UI update

### PR DESCRIPTION
**Pull request by chm-diederichs**
This PR updates the CLI flow and consolidates the response into a single message.

The updated flow is:

## generate-keys
```
> hypercore-sign-generate-keys

Your secret key will be encrypted with a password.
Please choose one now:

Password: 
Confirm password: 

Secret key written to /tmp.hypercore-sign/default
Public key written to /tmp/.hypercore-sign/default.public

Public key is xzub1puy9bm8m9mdr9x6p3n3w4pyramf4xt5381ugsky8q99atky
```

## sign
```
> hypercore-sign <request>

Request data:
{
  core: 'o6q5cw7p34eq6s4eqwyattucyumtkdragxeyxkcrekw6sd5zdq3y',
  fork: 0,
  length: 2,
  treeHash: '836fbcdb59e177e458e23b56cdb995d9e930ec3c17feb629db87f8592b566716'
}

Confirm? [y/N] yes

Request data is confirmed
Proceeding with signing...

Unlocking key pair...

Password: 

Signed with public key:

xzub1puy9bm8m9mdr9x6p3n3w4pyramf4xt5381ugsky8q99atky

Reply with:

qfzc7uucuypydk4rn5tdu3qfbykf1kcnsnnnemppfgm3w6sutdpz53o3g3oxoiui9it1xz9ghtc4pgoncf178a7hu3jumfydzx9heifbja5m8jfx8uja1gmqw1nyffjficwaza7mz9z8eerqhxbzqbt18ooaoownix9scposjxpi5o7j9wfp5d7o6hunfooaamjahqxc134ys
```

## verify
```
> hypercore-verify <response> <request> <publicKey>

Signature verified.

xzub1puy9bm8m9mdr9x6p3n3w4pyramf4xt5381ugsky8q99atky has signed the following request:
{
  core: 'o6q5cw7p34eq6s4eqwyattucyumtkdragxeyxkcrekw6sd5zdq3y',
  fork: 0,
  length: 2,
  treeHash: '836fbcdb59e177e458e23b56cdb995d9e930ec3c17feb629db87f8592b566716'
}
```

### CODE REVIEW

File: generate-keys.js [hunk 1]
```diff
@@ -8,23 +8,30 @@ const z32 = require('z32')
 
 const homeDir = os.homedir()
 
+// fs permissions
+const USER_ONLY_R = 0o400
+const USER_ONLY_RW = 0o600
+const USER_ONLY_RWX = 0o600
```
Comment (HDegroote) [inline generate-keys.js:14]: Should be 0o700 for RWX

File: generate-keys.js [hunk 2]
```diff
@@ -34,25 +41,21 @@ async function main () {
...
-  )
+  await fsProm.writeFile(publicKeyPath, z32.encode(publicKey), {
+    mode: USER_ONLY_RW
+  })
 
   // Prompt a confirmation when overwriting
   // (Because you probably don't want to overwrite these,
   // once they have been generated)
-  await fsProm.chmod(secretKeyLoc, 0o400)
+  await fsProm.chmod(secretKeyPath, USER_ONLY_R)
```
Comment (HDegroote) [inline generate-keys.js:55]: USER_ONLY_R  can actually be set when calling `writeFile` already, so no need to do it separately (hadn't realised when first writing this that I could set the mode there, so my bad)

File: lib/messages.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+const c = require('compact-encoding')
+
+const Response = {
+  preencode (state, res) {
```
Comment (HDegroote) [inline lib/messages.js:4]: Sanity check: no need to version this one in case we iterate on the format?

Comment (chm-diederichs) [reply lib/messages.js:6]: you're right, just added a version field

File: lib/secure.js [hunk 4]
```diff
@@ -193,7 +194,7 @@ function sign (data, keyBuffer, pwd) {
 }
 
 // function to accept password from user
-async function readPassword (prompt = 'Key-pair password:') {
+async function readPassword (prompt = 'Password: ') {
```
Comment (mafintosh) [inline lib/secure.js:197]: Can we still say Key-pair or similar here, as otherwise I kept thinking it was asking for my sudo password

**Pull request merged**